server/services/user.service.js

```diff
@@ -525,3 +525,44 @@ export const updateVendor = async ({ updatedVendor, vendorId }) => {
     throw new ErrorHandler(httpStatus.BAD_REQUEST, 'Error updating vendor information', error);
   }
 };
+
+export const addDirector = async ({ directorInfo, userId }) => {
+  const user = await getUserById(userId).catch((error) => {
+    throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+  });
+
+  try {
+    return User.findByIdAndUpdate(
+      user._id,
+      { $push: { 'vendor.directors': directorInfo } },
+      { new: true, fields: '-password' },
+    );
+  } catch (error) {
+    throw new ErrorHandler(httpStatus.BAD_REQUEST, 'Error adding director', error);
+  }
+};
+
+export const removeDirector = async ({ directorId, vendor }) => {
+  let { directors } = vendor.vendor;
+
+  directors = directors.filter((director) => director._id.toString() !== directorId.toString());
```

Comment:
I believe the two lines can be merged together into 1
```suggestion
  const signatories = directors.filter((director) => director._id.toString() !== directorId.toString() && director.isSignatory);
```

You can then proceed to check if there are signatories for verified vendors

```js
if (signatories.length === 0 && user.vendor.verified) {
...
```

Reply:
Feel free to ignore the above comment since you are replacing the whole director array (though I prefer using mongodb to remove it). However, you might want to make a few updates to prevent mutating the director array and making the code to make it more understandable:

```js
const { directors } = vendor.vendor;

  const updatedDirectors = directors.filter((director) => director._id.toString() !== directorId.toString());
...
```

Reply:
care to clarify on the ```js if (signatories.length === 0 && user.vendor.verified) ``` part and why adding `&& director.isSignatory`?

Reply:
Explained the reason here and told you to feel free to ignore it here
https://github.com/Highrachy/ballers-api/pull/94/files#r552943592